src/hamcrest/library/object/hasreturnvalue.py

```diff
@@ -0,0 +1,133 @@
+
+        if not hasattr(item, self.method_name):
+            return False
+
+        return_value = self.get_return_value(item)
+        return self.value_matcher.matches(return_value)
+
+    def describe_to(self, description: Description) -> None:
+        description.append_text("an object with a method '").append_text(
+            self.method_name
+        ).append_text("' whose return value when called with ").append_text(
+            self.get_callargs_str()
+        ).append_text(
+            " matches "
+        ).append_description_of(
+            self.value_matcher
+        )
+
+    def describe_mismatch(self, item: object, mismatch_description: Description) -> None:
+        if item is None:
+            mismatch_description.append_text("was None")
+            return
+
+        if not hasattr(item, self.method_name):
+            mismatch_description.append_description_of(item).append_text(
+                " did not have a "
+            ).append_description_of(self.method_name).append_text(" method")
+            return
+
+        mismatch_description.append_text("method ").append_description_of(
+            self.method_name
+        ).append_text(" called with ").append_text(self.get_callargs_str()).append_text(" ")
+
+        self.value_matcher.describe_mismatch(self.get_return_value(item), mismatch_description)
+
+    def __str__(self):
+        d = StringDescription()
+        self.describe_to(d)
+        return str(d)
+
+
+def has_return_value(
+    name: str,
+    match: Union[None, Matcher[V], V] = None,
+    args: Iterable[Any] = None,
+    kwargs: Mapping[str, Any] = None,
+) -> Matcher[object]:
+    """Matches if object has a method with a given name whose return value when
+    called with the given args and kwargs satisfies a given matcher.
+
+    :param name: The name of the method.
+    :param match: Optional matcher to satisfy.
+    :param args: Optional list of arguments to pass to the method.
+    :param kwargs: Optional mapping of keyword arguments to pass to the method.
+
+    This matcher determines if the evaluated object has a method with a given
+    name. If no such method is found, ``has_return_value`` is not satisfied.
+
+    If the method is found, it is called with the given arguments and keyword
+    arguments, and its return value is passed to a given matcher for 
+    evaluation. If the ``match`` argument is not a matcher, it is implicitly
+    wrapped in an :py:func:`~hamcrest.core.core.isequal.equal_to` matcher to
+    check for equality.
+
+    If the ``match`` argument is not provided, the
+    :py:func:`~hamcrest.core.core.isnone.not_none` matcher is used so that
+    ``has_return_value`` is satisfied if a matching method is found which
+    returns something.
+
+    Examples::
+
+        has_return_value('is_displayed', True)
+        has_return_value('get_child', is_instance_of(Child))
+        has_return_value('name')
+
+    """
+
+    if match is None:
+        match = not_none()
```

Comment:
Feels like an Anything matcher would be less surprising here.

Reply:
Doesn't `is_anything` match on `None`, though? 

I had `is_anything` here before, but matching on `None` was my main concern. I thought that `has_return_value` should complain if the method it's testing returns `None`, since that would mean it doesn't have a return value. 

I'm willing to be dissuaded, though, especially because changing to `is_anything` would mean avoiding the MyPy fixes i mentioned in the other comment.